### PR TITLE
[WIP] Supersonic inlet profile

### PR DESCRIPTION
**Pull request by eskimmel**
## Proposed Changes
*Give a brief overview of your contribution here in a few sentences.*
This is a draft pull request for the implementation of a supersonic inlet profile feature. So far, the only changes made have been in CEulerSolver.cpp, where the uniform temperature, pressure, and velocity at the inlet for a supersonic inlet BC have been replaced with vertex values (please verify I did this correctly).

## Related Work
*Resolve any issues (bug fix or feature request), note any related PRs, or mention interactions with the work of others, if any.*
I have been in communication with @bigfooted regarding the first steps for this implementation. I created a feature request for this on 5/25/2022.

## PR Checklist
*Put an X by all that apply. You can fill this out after submitting the PR. If you have any questions, don't hesitate to ask! We want to help. These are a guide for you to know what the reviewers will be looking for in your contribution.*

- [X] I am submitting my contribution to the develop branch.
- [ ] My contribution generates no new compiler warnings (try with the '-Wall -Wextra -Wno-unused-parameter -Wno-empty-body' compiler flags, or simply --warnlevel=2 when using meson).
- [ ] My contribution is commented and consistent with SU2 style.
- [ ] I have added a test case that demonstrates my contribution, if necessary.
- [ ] I have updated appropriate documentation (Tutorials, Docs Page, config_template.cpp) , if necessary.

### CODE REVIEW

**Review by pcarruscag — COMMENTED** [review]
Actually this is more involved than I thought.
If you can attend the developer meeting next Tuesday at 10am Pacific time (https://meet.jit.si/SU2_DevMeeting) I can try to give directions.
Otherwise I suggest that you avoid intersections between supersonic inlets and walls.

File: SU2_CFD/src/solvers/CEulerSolver.cpp [hunk 1]
```diff
@@ -2524,6 +2524,7 @@ void CEulerSolver::GetPower_Properties(CGeometry *geometry, CConfig *config, uns
     su2double *Inlet_XCG              = new su2double [config->GetnMarker_All()]();
     su2double *Inlet_YCG              = new su2double [config->GetnMarker_All()]();
     su2double *Inlet_ZCG              = new su2double [config->GetnMarker_All()]();
+    su2double *Inlet_Velocity         = new su2double [config->GetnMarker_All()]();
```
[the remark this answers is not in the document]
Comment (eskimmel) [reply SU2_CFD/src/solvers/CEulerSolver.cpp:2527]: I'll attend the meeting. Upon thinking about it more, I think I will be breaking my inlet into two domains, one that uses the subsonic inlet feature (directly above the wall) and one that would use the new supersonic inlet feature (above the subsonic region). That way, the supersonic inlet profile will not be meeting the no-slip wall and contain subsonic flow.

**Comment by eskimmel** [comment]
I just committed a second round of changes that I would appreciate some feedback on. Compilation is successful with these changes, however, upon testing, I receive the following message: 

Error in "void CConfig::SetConfig_Parsing(std::istream&)": 
-------------------------------------------------------------------------
Line 271 SPECIFIED_SUPERSONIC_INLET_PROFILE: invalid option name. Check current SU2 options in config_template.cfg.
Did you mean SPECIFIED_INLET_PROFILE?
Line 274 SUPERSONIC_INLET_FILENAME: invalid option name. Check current SU2 options in config_template.cfg.
Did you mean INLET_FILENAME?

I had already adjusted CConfig.cpp to include supersonic inlet profile inputs, but apparently these are not being read. Any suggestions?

**Comment by eskimmel** [comment]
Whoops, thanks for letting me know! I'll fix that shortly.

**Comment by bigfooted** [comment]
OK, great. 
In the mean time, some changes were made to CEulerSolver.cpp by @pcarruscag , which is why it now says that there is a merge conflict here. You'll have to put these changes into your CEulerSolver.cpp to get rid of this conflict.

**Comment by eskimmel** [comment]
Upon implementing these changes, the compilation of the code failed due to multiple variables receiving conflicting declarations. After using just the variables @pcarruscag made and re-compiling, I receive multiple errors regarding the assignment of read-only variables, so I had to remove the "const" before a few lines to allow for compilation. Is this alright?

Also, any ideas about why SPECIFIED_SUPERSONIC_INLET_PROFILE is not being accepted as an option even though it's defined in CConfig.cpp?

**Comment by pcarruscag** [comment]
Why did you introduce supersonic-specific options? The inlet file allows specifying profiles for all markers, and a marker will not have 2 boundary conditions. From what I saw we just need to interpret the values in a different way.

**Comment by eskimmel** [comment]
I introduced supersonic specific options because, from my understanding, anything written in INLET_FILENAME in the .cfg file will be treated as a subsonic inlet, even if the marker tag is assigned to a supersonic inlet. I wanted the static temperature, static pressure, and velocity vector to be read into SU2 for the supersonic profile and the total temperature, total pressure, and unit flow vector to be read for the subsonic profile. I figured that placing these different values in different files would be the easiest way to circumvent the issue stated above.

**Comment by pcarruscag** [comment]
I see, that should not be necessary because internally we can check that the marker is supersonic instead of subsonic and give the values a different interpretation.
The incompressible solver does this when the inlet type is changed from total conditions to velocity inlet.

File: SU2_CFD/src/solvers/CEulerSolver.cpp [hunk 2]
```diff
@@ -9427,3 +9437,4 @@ void CEulerSolver::GatherInOutAverageValues(CConfig *config, CGeometry *geometry
     }
   }
 }
+
```
Comment (WallyMaier) [inline SU2_CFD/src/solvers/CEulerSolver.cpp:9440]:
```suggestion
```

**Pull request closed without merging**